java-frontend/src/test/files/se/StreamNotConsumedCheck.java

```diff
@@ -63,4 +63,55 @@ void testField() {
     this.streamField = streamField.filter(e -> true); // Compliant, assigned to field
   }
 
+  void exceptionalPath() {
+    try {
+      Stream.of(1,2,3)
+        .filter(e -> true)
+        .count();
+    } catch (Exception ex) {
+
+    }
+  }
+
+  void exceptionalPath2() {
+    try {
+      Stream.of(1,2,3).filter(e -> true);  // Noncompliant
+
+      Stream.of(1,2,3)
+        .filter(e -> true)
+        .skip(skipCount())
+        .count();
+
+      Stream.of(1,2,3).skip(skipCount()); // Noncompliant
+    } catch (Exception ex) {
+      System.out.println("Exception!");
+    }
+  }
+
+  void exceptionalPathFN(Stream<Object> stream) {
+    try {
+      stream = stream.skip(skipCount());  // Compliant this pipeline is not consumed on exceptional path
```

Comment:
comment is confusing with the method name : is that a false negative ? if yes, state it in the comment.